pyads/symbol.py

```diff
@@ -0,0 +1,267 @@
+# this 'if' to only include it when type hinting (False during execution)
+if TYPE_CHECKING:
+    from .ads import Connection  # pragma: no cover
+
+
+class AdsSymbol:
+    """Object that points to an ADS variable
+
+    Contains index group, index offset, name, symbol type, comment of ADS
+    symbol. Also remembers a reference to a Connection to be able to
+    read/write directly.
+
+    The virtual property `value` can be used to read from and write to
+    the symbol.
+
+    :ivar index_group: Index group of symbol
+    :ivar index_offset: Index offset of symbol
+    :ivar name: Name of symbol
+    :ivar symbol_type: String representation of symbol type (PLC-style,
+                     e.g. "LREAL")
+    :ivar plc_type: ctypes type of variable (from constants.PLCTYPE_*)
+    :ivar comment: Comment of symbol
+    """
+
+    def __init__(self,
+                 plc: "Connection",
+                 name: Optional[str] = None,
+                 index_group: Optional[int] = None,
+                 index_offset: Optional[int] = None,
+                 symbol_type: Optional[str] = None,
+                 comment=None):
+        """Create AdsSymbol instance
+
+        Specify either the variable name or the index_group **and**
+        index_offset so the symbol can be located.
+        If the name was specified but not all other attributes were,
+        the other attributes will be looked up from the connection.
+        `symbol_type` should be a string representing a PLC type (e.g.
+        'LREAL').
+
+        :param plc: Connection instance
+        :param name:
+        :param index_group:
+        :param index_offset:
+        :param symbol_type: PLC variable type (e.g. 'LREAL')
+        :param comment:
+        """
+        self._plc = plc
+
+        self._handles_list: List[Tuple[int, int]] = []  # Notification handles
+
+        do_lookup = True
+
+        if index_group is None or index_offset is None or symbol_type is None:
+            if name is None:
+                raise ValueError('Please specify either `name`, or '
+                                 '`index_group`, `index_offset` and '
+                                 'plc_type')
+        else:
+            # We have an address and the type, so we don't need to do a lookup
+            do_lookup = False
+
+        self.name = name
+        self.index_offset = index_offset
+        self.index_group = index_group
+        self.symbol_type = symbol_type
+        self.comment = comment
+
+        if do_lookup:
+            self.make_symbol_from_info()  # Perform remote lookup
+
+        # Now `self._type_hint` must have a value, find the actual PLCTYPE
+        # from it.
+        # This is relevant for both lookup and full user definition.
+
+        if self.symbol_type is not None:
+            self.plc_type = self.get_type_from_str(self.symbol_type)
+        else:
+            # Set directly from user input
+            self.plc_type = self.symbol_type  # type: Any
```

Comment:
So basically you set plc_type to None if no symbol_type was provided, otherwise you try to get the type?

```suggestion
        self.plc_type: Optional[Any] = None
        if self.symbol_type is not None:
            self.plc_type = self.get_type_from_str(self.symbol_type)
```